verilog/tools/lint/lint_tool_test.sh

```diff
@@ -306,4 +306,496 @@ grep -q "module-parameter" "${MY_OUTPUT_FILE}.err" || {
 }
 
 ################################################################################
+echo "=== Test --autofix=yes"
+
+ORIGINAL_TEST_FILE="${TEST_TMPDIR}/autofix_original.sv"
+REFERENCE_TEST_FILE="${TEST_TMPDIR}/autofix_reference.sv"
+TEST_FILE="${TEST_TMPDIR}/autofix.sv"
+
+# Fixable violations, in order:
+# :2:10: no-trailing-spaces
+# :3:10: forbid-consecutive-null-statements
+# :4:10: forbid-consecutive-null-statements
+# :4:11: no-trailing-spaces
+# :5:10: forbid-consecutive-null-statements
+# :6:10: forbid-consecutive-null-statements
+# :7:10: no-trailing-spaces
+# :7:14: posix-eof
+>"${ORIGINAL_TEST_FILE}"
+echo -en 'module Autofix;    \n' >>"${ORIGINAL_TEST_FILE}"
+echo -en '  wire a;;\n'          >>"${ORIGINAL_TEST_FILE}"
+echo -en '  wire b;;  \n'        >>"${ORIGINAL_TEST_FILE}"
+echo -en '  wire c;;\n'          >>"${ORIGINAL_TEST_FILE}"
+echo -en '  wire d;;\n'          >>"${ORIGINAL_TEST_FILE}"
+echo -en 'endmodule    '         >>"${ORIGINAL_TEST_FILE}"
+>"${REFERENCE_TEST_FILE}"
+echo -en 'module Autofix;\n' >>"${REFERENCE_TEST_FILE}"
+echo -en '  wire a;\n'       >>"${REFERENCE_TEST_FILE}"
+echo -en '  wire b;\n'       >>"${REFERENCE_TEST_FILE}"
+echo -en '  wire c;\n'       >>"${REFERENCE_TEST_FILE}"
+echo -en '  wire d;\n'       >>"${REFERENCE_TEST_FILE}"
+echo -en 'endmodule\n'       >>"${REFERENCE_TEST_FILE}"
+
+cp "${ORIGINAL_TEST_FILE}" "${TEST_FILE}"
+
+RULES_CONFIG_FILE="${TEST_TMPDIR}/rules.conf"
+cat > ${RULES_CONFIG_FILE} <<EOF
+forbid-consecutive-null-statements
+no-trailing-spaces
+posix-eof
+EOF
+
+"$lint_tool" --ruleset=none --rules_config="${RULES_CONFIG_FILE}" --autofix=yes \
+    "${TEST_FILE}" > /dev/null 2>&1
+
+check_diff() {
+  local reference_file_="$1"
+  local test_file_="$2"
+  local diff_file_="$3"
+  local err_message_="$4"
+
+  local vis_reference_file_="${reference_file_}.vis"
+  local vis_test_file_="${test_file_}.vis"
+
+  # Show invisible characters
+  local invisible_chars_re_='s/ /·/g;s/\t/├───/g;s/\r/⁋/g;s/\n/¶\n/g'
+  sed -z "$invisible_chars_re_" "$reference_file_" > "$vis_reference_file_"
```

Comment:
`-z` is a GNU extension and will not necessarily work on various platforms.
Can you reformulate it so that we can use a standard `sed` ?

Reply:
Should be OK now